### ISSUE TITLE
[Bug]: Text selection via accessibility on macOS broken when lines start with blank

### ISSUE THREAD

**Issue by greimers** [electron/electron#36337]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

Unknown / Latest

### What operating system are you using?

macOS

### Operating System Version

macOS Ventura

### What arch are you using?

arm64 (including Apple Silicon)

### Expected Behavior

I am working on a (native) macOS app that needs to select text in other apps using [macOS Accessibility API ](https://developer.apple.com/library/archive/documentation/Accessibility/Conceptual/AccessibilityMacOSX/index.html).

The following Swift code should select the given range of characters in a text field:

```
let element: AXUIElement = someElement
var range = CFRange(location: 1, length: 3)
if let axRangeValue: AXValue = AXValueCreate(AXValueType.cfRange, &range)  {
    AXUIElementSetAttributeValue(
      element,
      kAXSelectedTextRangeAttribute as CFString,
      axRangeValue
  )
}
```

### Actual Behavior

In Electron apps (e.g. Slack and Obsidian) it generally works but it is broken in some cases:

* when a line of text starts with one or more blanks, the Electron app will select the wrong character range (one char missing)
* when the range is exactly the last characters of a line, the Electron app will also select the following line break even though it is outside the given range

These issues do not arise in native Mac apps.

That Electron apps get confused with the selection when a line starts with blank can easily be seen in the Accessibility Inspector app (comes with XCode).  In the following screenshot from Obsidian, the word "This" is selected. But the accessibility inspector shows that the text field reports a wrong selection range of (loc: 2, length: 4). Correct would be (loc: 1, length: 4)

<img width="675" alt="Screenshot 2022-11-14 at 18 09 20" src="https://user-images.githubusercontent.com/5694277/201725988-4c8fbde4-cadc-4781-b3e5-5dbe7c01f188.png">

Please note that inspecting Electron apps with the Accessibility Inspector requires [programmatically activating accessibility access](https://www.electronjs.org/docs/latest/tutorial/accessibility#macos) for them. My sample project covers that.

### Additional Information

I have created a simple sample app in Swift that can be used to reproduce the issue and play with text selection using Accessibility API. It includes all the required Accessibility calls and a simply UI to trigger text selection events on the currently focused app (can be any app).

The Zip contains an XCode SwiftUI project that is ready to run and has a ReadMe with instructions to reproduce the bugs.

[Electron Accessibilty Test.zip](https://github.com/electron/electron/files/10005544/Electron.Accessibilty.Test.zip)

**Comment by mlaurencin** [comment]
@greimers Are you able to reproduce this issue in Chrome?

**Comment by greimers** [comment]
@mlaurencin Well, partially.
Programmatic selection using Accessibility is broken in Chrome as well. (When text starts with a blank the selected text will be one-off in length).
However, Chrome shows the correct selection ranges for selected text in the Accessibility Inspector.

**Comment by vojtabohm** [comment]
Is there any progress on this issue? I am developing a similar app as the author and I have stumbled across this bug as well and unfortunately it makes my app unusable in these cases.

**Comment by greimers** [comment]
bump

**Comment by greimers** [comment]
This is still an issue

**Comment by greimers** [comment]
Still an issue.

**Comment by greimers** [comment]
bump

**Comment by greimers** [comment]
bump

**Comment by Madd0g** [comment]
suffering from this issue with apps like Obsidian - with multi-line text the problem seems worse and what is shown as selected is very unpredictable.

also, activating accessibility programmatically works on one computer, but not the other - maybe there's also an additional setting I need to turn on per app?

**Comment by Madd0g** [comment]
"bump"

**Comment by idokraicer** [comment]
any fix to this issue?
